siriuspy/siriuspy/injctrl/main.py

```diff
@@ -574,7 +574,7 @@ def set_bucketlist_step(self, step):
                 return False
             bucket = _np.arange(self._topupnrpulses) + 1
             bucket *= step
-            bucket += self._evg_dev.bucketlist[0] - 1
+            bucket += self._evg_dev.bucketlist_mon[-1] - 1
```

Comment:
@anacso17 I thnk you need to test whether the value in the bucketlist is valid or not before using it. Check if it is a number in [1, #harm] and in case it is not use 1.

Reply:
done!